apps/extension/src/App/Accounts/AddAccount.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button, ButtonVariant, Input, InputVariants } from "@anoma/components";
+
+import { ExtensionRequester } from "extension";
+import { Ports } from "router";
+import { DerivedAccount, DeriveAccountMsg } from "background/keyring";
+import {
+  AddAccountContainer,
+  AddAccountForm,
+  Bip44PathContainer,
+  Bip44PathDelimiter,
+  Bip44Input,
+  Bip44Path,
+  Bip44Error,
+  ButtonsContainer,
+  FormError,
+  FormStatus,
+  InputContainer,
+  Label,
+} from "./AddAccount.components";
+
+type Props = {
+  accounts: DerivedAccount[];
+  requester: ExtensionRequester;
+  setAccounts: (accounts: DerivedAccount[]) => void;
+};
+
+const validateAccount = (
+  newAccount: { account: number; change: number; index: number },
+  accounts: DerivedAccount[]
+): boolean => {
+  const newPath = [
+    newAccount.account,
+    newAccount.change,
+    newAccount.index,
+  ].join("/");
+  let isValid = true;
+  accounts.forEach((a: DerivedAccount) => {
+    const { bip44Path } = a;
+    const { account, change, index } = bip44Path;
+    const existingPath = [account, change, index].join("/");
+
+    if (newPath === existingPath) {
+      isValid = false;
+    }
+  });
+
+  return isValid;
+};
+
+const findNextIndex = (accounts: DerivedAccount[]): number => {
+  let maxIndex = 0;
+  accounts.forEach((account) => {
+    const { index } = account.bip44Path;
+    if (index > maxIndex) {
+      maxIndex = index;
+    }
+  });
+
+  return maxIndex + 1;
+};
+
+enum Status {
+  Pending,
+  Failed,
+}
+
+const AddAccount: React.FC<Props> = ({ accounts, requester, setAccounts }) => {
+  const navigate = useNavigate();
+  const [alias, setAlias] = useState("");
+  const [account, setAccount] = useState(0);
+  const [change, setChange] = useState(0);
+  const [index, setIndex] = useState(findNextIndex(accounts));
+  const [bip44Error, setBip44Error] = useState("");
+  const [isFormValid, setIsFormValid] = useState(true);
+  const [formError, setFormError] = useState("");
+  const [formStatus, setFormStatus] = useState<Status>();
```

Comment:
nit: I am wondering if we should have also Idle. As now we have 4 states, `undefined`, `Pending`, `Failed`. We have no `Completed` but this component disappears as soon as it it success?